packages/react/src/index.tsx

```diff
@@ -11,6 +11,10 @@ export * as primitives from './primitives';
 export { theme, translations } from '@aws-amplify/ui';
 
 export {
+  /**
+   * @deprecated `AmplifyAuthenticator` is the v1 release. Prefer `Authenticator` for continued support.
+   */
+  AmplifyAuthenticator,
   AmplifyChatbot,
   AmplifyPhotoPicker,
   AmplifyPicker,
```

Comment:
All legacy packages could be re-exported from `@aws-amplify/ui-react/legacy`.

Reply:
I'm in favor of exporting from an explicit `legacy` path.

Reply:
I also like the idea to keep it as separated as possible, but IMO that could break existing users that rely on AmplifyAuthenticator (like customers depending on `@aws-amplify/ui-react`: "latest"`.

Is there a export symbol name conflict that would force us to do it? Or tree-shaking considerations?

Reply:
`@aws-amplify/ui-react@2` is already going to be a breaking change:

- `Authenticator` is new whereas the previous version had the prefix `AmplifyAuthenticator`.
- Plus, `withAuthenticator` isn't backwards-compatible (`usernameAlias` and `formFields` are gone).
- Neither are our CSS variables for theming.

For customers who use `latest` (which would be a bit rare, since npm & yarn both resolve to version number (plus lock files prevent this):

```console
$ npm install @aws-amplify/ui-react@latest --save
```

```json
  "dependencies": {
    "@aws-amplify/ui-react": "^1.2.14"
  }
```

Reply:
If we don't need to do this, then we don't have to.  This was something I stumbled upon since we're already re-exporting `AmplifyS3Image` for customers to use alongside the new Authenticator.

But I really like the idea of re-exporting under `/legacy` because it's near 0 cost to do so and we can continue creating net-new components/primitives without worrying about "cleaning up" our exports.

Reply:
I like this `/legacy` . It will break in 2.0 but most people should expect that.

Reply:
thanks for providing so much information & context to understand the issue, I'm convinced now about using `/legacy` now. Are you going to include that change on this PR or will create a separate one?

Reply:
Included & updated!